src/SolAR3DPointsViewerOpengl.cpp

```diff
@@ -329,12 +369,28 @@ void SolAR3DPointsViewerOpengl::OnRender()
         glPointSize(m_pointSize);
         glBegin(GL_POINTS);
         for (unsigned int i = 0; i < m_points.size(); ++i) {
-         if (m_fixedPointsColor)
-            glColor3f(m_pointsColor[0], m_pointsColor[1], m_pointsColor[2]);
-         else
-             glColor3f(m_points[i]->getR(), m_points[i]->getG(), m_points[i]->getB());
-
-         glVertex3f(m_points[i]->getX(), -m_points[i]->getY(), -m_points[i]->getZ());
+            // if color map is provided, display point cloud according to class colors
```

Comment:
Looks similar to code above, is it possible to factorize it in a function/lambda ?